app/assets/javascripts/components/build_graph.es6.jsx

```diff
@@ -0,0 +1,118 @@
+    this.setState({ simplificationTolerance: parseInt(simplificationTolerance, 10) });
+  }
+
+  handleFormatChange(event) {
+    this.setState({ format: event.target.value });
+  }
+
+  render() {
+    if (!this.props) {
+      return null;
+    }
+    return (
+      <div>
+        <div className="form-group row">
+          <label
+            className="col-sm-3 form-text text-capitalize"
+            data-toggle="tooltip"
+            htmlFor="simplification-method"
+          >
+            Simplification Method
+          </label>
+
+          <div className="col-sm-6">
+            <select
+              name="collection_template[graph][simplification-method]"
+              value={this.state.simplificationMethod}
+              onChange={this.handleMethodChange}
+              className={'form-control'}
+            >
+              { this.props.simplificationMethods.map(method =>
+                <option value={method.toLowerCase()} key={Math.random()}>{method}</option>,
+                )}
+            </select>
+          </div>
+        </div>
+        <div className="form-group row">
+          <label
+            className="col-sm-3 form-text text-capitalize"
+            data-toggle="tooltip"
+            htmlFor="simplificationTolerance"
+          >Simplification Tolerance
+          </label>
+          <div className="col-sm-6">
+            <Range
+              fieldName="collection_template[graph][simplification-tolerance]"
+              value={this.state.simplificationTolerance}
+              enabled
+              onUpdate={this.handleToleranceChange}
+              className="align-middle"
+              max={10}
+            />
+          </div>
+          <div className="col-sm-2">
+            {this.state.simplificationTolerance}
+          </div>
+        </div>
+        <div className="form-group row">
+          <label
+            className="col-sm-3 form-text text-capitalize"
+            data-toggle="tooltip"
+            htmlFor="format"
+          >
+            Graph format
+          </label>
+          <div className="col-sm-6">
+            <select
+              name="collection_template[graph][format]"
+              value={this.state.format}
+              onChange={this.handleFormatChange}
+              className={'form-control'}
+            >
+              { this.props.formats.map(format =>
+                <option value={format.toLowerCase()} key={Math.random()}>{format}</option>,
+                )}
+            </select>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+Graph.propTypes = {
+  simplificationMethod: React.PropTypes.string,
+  simplificationMethods: React.PropTypes.arrayOf(React.PropTypes.string),
+  simplificationTolerance: React.PropTypes.string,
+  format: React.PropTypes.string,
+  formats: React.PropTypes.arrayOf(React.PropTypes.string),
+};
+
+Graph.defaultProps = {
+  simplificationMethod: 'vw',
+  simplificationMethods: ['vw', 'rdp'],
```

Comment:
It might be a bit less cryptic to have the actual names of the methods being displayed ('rdp' stands for Ramer–Douglas–Peucker's algorithm, and 'vw' for Visvalingam–Whyatt's algorithm).